mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/mail/dao/PostgresMailboxDAOImpl.java

```diff
@@ -0,0 +1,165 @@
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.jpa.mail.dao;
+
+import static org.apache.james.mailbox.jpa.mail.PostgresMailboxTable.MAILBOX_ID;
+import static org.apache.james.mailbox.jpa.mail.PostgresMailboxTable.MAILBOX_NAME;
+import static org.apache.james.mailbox.jpa.mail.PostgresMailboxTable.MAILBOX_NAMESPACE;
+import static org.apache.james.mailbox.jpa.mail.PostgresMailboxTable.MAILBOX_UID_VALIDITY;
+import static org.apache.james.mailbox.jpa.mail.PostgresMailboxTable.TABLE_NAME;
+import static org.apache.james.mailbox.jpa.mail.PostgresMailboxTable.USER_NAME;
+import static org.jooq.impl.DSL.count;
+
+import java.util.NoSuchElementException;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.exception.MailboxExistsException;
+import org.apache.james.mailbox.jpa.PostgresMailboxId;
+import org.apache.james.mailbox.jpa.mail.PostgresMailboxTable;
+import org.apache.james.mailbox.model.Mailbox;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.UidValidity;
+import org.apache.james.mailbox.model.search.MailboxQuery;
+import org.apache.james.mailbox.store.MailboxExpressionBackwardCompatibility;
+import org.jooq.Record;
+import org.jooq.Record1;
+
+import com.google.common.base.Preconditions;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresMailboxDAOImpl implements PostgresMailboxDAO {
+    private static final char SQL_WILDCARD_CHAR = '%';
+    private static final String DUPLICATE_VIOLATION_MESSAGE = "duplicate key value violates unique constraint";
+
+    private final PostgresExecutor postgresExecutor;
+
+    @Inject
+    public PostgresMailboxDAOImpl(PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    @Override
+    public Mono<Mailbox> create(MailboxPath mailboxPath, UidValidity uidValidity) {
+        return postgresExecutor.dslContext()
+            .flatMapMany(dslContext -> Mono.from(dslContext.insertInto(TABLE_NAME, MAILBOX_ID, MAILBOX_NAME, USER_NAME, MAILBOX_NAMESPACE, MAILBOX_UID_VALIDITY)
+                .values(PostgresMailboxId.generate().asUuid(), mailboxPath.getName(), mailboxPath.getUser().asString(), mailboxPath.getNamespace(), uidValidity.asLong())
```

Comment:
Allocate the postgresId to a variable and reuse it?
THis might require less network and less parsing...

Reply:
less network??